src/entities/purpleAddress.ts

```diff
@@ -0,0 +1,37 @@
+import {
+  BaseEntity,
+  Column,
+  Entity,
+  Index,
+  ManyToOne,
+  PrimaryGeneratedColumn,
+  RelationId,
+} from 'typeorm';
+import { Field, ID, ObjectType } from 'type-graphql';
+import { Project } from './project';
+import { User } from './user';
+
+@Entity()
+@ObjectType()
+export class PurpleAddress extends BaseEntity {
+  @Field(type => ID)
+  @PrimaryGeneratedColumn()
+  readonly id: number;
+
+  @Index()
+  @Field({ nullable: true })
+  @Column({ nullable: true })
+  description?: string;
+
+  @Index()
+  @Field()
+  @Column({ unique: true })
+  address: string;
+
+  @Index()
+  @Field(type => Project)
+  @ManyToOne(type => Project, { eager: true })
+  project: Project;
+  @RelationId((purpleAddress: PurpleAddress) => purpleAddress.project)
+  projectId: number;
```

Comment:
I have a question, we have the projectId here but we are not using it on the queries below. 
This is only to track from what project is the address right? Because a purple list address is banned from getting givBacks from what I read on the issues you linked.

Reply:
Or should we use the projectId for filtering? I don't so tho.

Reply:
@CarlosQ96 We dont use projectId in our query, but it's can be viewed in admin-bro, so admins can know this purple address is related to what project, or they can see the purple addresses of a project in admin-bro, so I think it's useful to have it, we can manage purple addresses easier in this way

Reply:
Great, Thanks! I think you can merge!

Reply:
And all purple addresses are related to some projects, so it's better to reflect that relationship in our DB